doc/examples/uuid.rst

```diff
@@ -0,0 +1,472 @@
+  uuid_obj = uuid4()
+
+  # save the native uuid object to MongoDB
+  collection.insert_one({'uuid': uuid_obj})
+
+  # retrieve the stored uuid object from MongoDB
+  document = collection.find_one({})
+
+  # check that the retrieved UUID matches the inserted UUID
+  assert document['uuid'] == uuid_obj
+
+Native :class:`uuid.UUID` objects can also be used as part of MongoDB
+queries::
+
+  document = collection.find({'uuid': uuid_obj})
+  assert document['uuid'] == uuid_obj
+
+The above examples illustrate the simplest of use-cases - one where the
+UUID is generated by, and used in the same application. However,
+the situation can be significantly more complex when dealing with a MongoDB
+deployment that contains UUIDs created by other drivers as the Java and CSharp
+drivers have historically encoded UUIDs using a byte-order that is different
+from the one used by PyMongo. Applications that require interoperability across
+these drivers must specify the appropriate
+:class:`~bson.binary.UuidRepresentation`.
+
+In the following sections, we describe how drivers have historically differed
+in their encoding of UUIDs, and how applications can use the
+:class:`~bson.binary.UuidRepresentation` configuration option to maintain
+cross-language compatibility.
+
+.. attention:: Applications that do not share a MongoDB deployment with
+   any other application and that have never stored UUIDs in MongoDB
+   should use the ``standard`` UUID representation for cross-language
+   compatibility. See :ref:`configuring-uuid-representation` for details
+   on how to configure the :class:`~bson.binary.UuidRepresentation`.
+
+.. _example-legacy-uuid:
+
+Legacy Handling of UUID Data
+----------------------------
+
+Historically, MongoDB Drivers have used different byte-ordering
+while serializing UUID types to :class:`~bson.binary.Binary`.
+Consider, for instance, a UUID with the following canonical textual
+representation::
+
+  00112233-4455-6677-8899-aabbccddeeff
+
+This UUID would historically be serialized by the Python driver as::
+
+  00112233-4455-6677-8899-aabbccddeeff
+
+The same UUID would historically be serialized by the C# driver as::
+
+  33221100-5544-7766-8899-aabbccddeeff
+
+Finally, the same UUID would historically be serialized by the Java driver as::
+
+  77665544-3322-1100-ffee-ddccbbaa9988
+
+.. note:: For in-depth information about the the byte-order historically
+   used by different drivers, see the `Handling of Native UUID Types
+   Specification
+   <https://github.com/mongodb/specifications/blob/master/source/uuid.rst>`_.
+
+This difference in the byte-order of UUIDs encoded by different drivers can
+result in highly unintuitive behavior in some scenarios. We detail two such
+scenarios in the next sections.
+
+Scenario 1: Applications Share a MongoDB Deployment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Consider the following situation:
+
+* Application ``M`` written in C# generates a UUID and uses it as the ``_id``
```

Comment:
Can you change `M` and `N` to `A` and `B` (or `C` for C# and `P` for Python)? `M` and `N` sound and look very similar.

Reply:
Done. Used C and P